src/state/shellscripts/bwrap.sh

```diff
@@ -49,16 +54,43 @@ add_ccache_mount() {
 COMMAND="$1"; shift
 case "$COMMAND" in
     build)
+        # mount unusual path in ro
+        if  [ -n "${OPAM_USER_PATH_RO-}" ]; then
+           add_mounts ro $(echo ${OPAM_USER_PATH_RO} | sed 's|:| |g')
+        fi
+        #
+        # Some modules generates an error if the complete infra of {opam_root}
+        # is not seen in the sandbox
+        # Some modules want to write in {opam_root}/log/ while building.
+        #     [ERROR] /work/CG-2.1/FilteringProducts/_build/_opam exists, but does not appear to be a valid opam root.
+        #     Please remove it and use `opam init', or specify a different `--root' argument
+        # They also need to write in $(dirname $OPAM_SWITCH_PREFIX)/log
+        #  => Need to mount $(dirname $OPAM_SWITCH_PREFIX)/log in rw
+        add_mounts rw "$(dirname $OPAM_SWITCH_PREFIX)/log"
+        add_mounts ro "$(dirname $OPAM_SWITCH_PREFIX)/repo"
+        add_mounts ro "$(dirname $OPAM_SWITCH_PREFIX)/config"
+        add_mounts ro "$(dirname $OPAM_SWITCH_PREFIX)/config.lock"
+        add_mounts ro "$(dirname $OPAM_SWITCH_PREFIX)/download-cache"
+        add_mounts ro "$(dirname $OPAM_SWITCH_PREFIX)/lock"
+        add_mounts ro "$(dirname $OPAM_SWITCH_PREFIX)/opam-init"
```

Comment:
I'm afraid there is some confusion between `OPAMROOT` which contains [global configuration files](http://opam.ocaml.org/doc/2.0/Manual.html#opam-root) and `OPAM_SWITCH_PREFIX` which contains [local switch configuration and installation structure](http://opam.ocaml.org/doc/2.0/Manual.html#Structure). It's the opam root that contains those files, not the switch prefix. Plus, at build stage, opam don't need to access to its opam root (if so, it's a [packaging issue](https://github.com/ocaml/opam/issues/3434)).
I've checked and I can confirm that it works as is: having opam root and opam switch in an unlisted directory, launching an install from another one unrelated.
What is you configuration? You can have internal values of root & switch with `opam config list`.

Reply:
Ok
Seems to be package issue.
Shall I do another pr without this part of the patch ?

Reply:
> Seems to be package issue.

What is the package you tried to install?

> Shall I do another pr without this part of the patch ?

No need to create a new one, you can update this PR by updating the branch you used. You can squash commits in order to have a clean history.

Reply:
I've such issue with package cpuid  @ version 0.1.1.

I'll update the pr asap

Reply:
Usually, cpuid can be installed as `$HOME` is ro mounted and it uses opam to access configuration (via `ocb-stubblr`, cf. above links). The `ocb-stubblr` PR is not yet merged fixing it, so meanwhile it still need to access opam root, but just in ro.
I tried using the new `OPAM_USER_PATH_RO` variable set to an unusual opam root and it installed with no problem.

Reply:
Understood...
PR has been updated. It now only contains the OPAM_USER_PATH_RO patch